document-signing-certificate-issuer/template.yaml

```
@@ -243,6 +256,25 @@ Resources:
       Description: !Sub 'Example Document Signing Key 1 for ${AWS::StackName}'
       Enabled: true
       PendingWindowInDays: 7
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: EnableAccountRoot
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: 'kms:*'
+            Resource: '*'
+
+          - Sid: AllowWalletIssuerRoleToSign
+            Effect: Allow
+            Principal:
+              AWS: arn:aws:iam::921104553881:role/wallet-cred-issuer-be-WalletBEECSTaskRole-bSAppwk9KqvA
+            Action:
+              - kms:Sign
+              - kms:GetPublicKey
+              - kms:DescribeKey
```

```suggestion
              - kms:GetPublicKey
```

Can we double check that this action is needed?

---

I wasn't sure if we fetch key metadata before calling Sign, so i included this.

---

I realise this is redundant as we have GetPublicKey. I remove it.

---

Sorry, I've just noticed my suggested change was wrong. What I wanted to check is if we need `kms:GetPublicKey`. We do need `kms:DescribeKey` for sure. 

In any case, we should only grant the permissions that are actually needed ("least privilege"). Can we please check if `kms:GetPublicKey` is needed?